### PR TITLE
Alertmanager: Fix goroutine leak when stored config fails to apply and there is no existing tenant alertmanager

### PR DESCRIPTION
**Pull request by alexweav**
#### What this PR does

When a tenant has bad config in the store and no prior running alertmanager, we start a new one and try to apply the bad config to it. When the apply fails, we forget to shut down the alertmanager we just started.

This means we leak an alertmanager every time we sync that user's config.

There is a check for config correctness earlier on when syncing configs, but it is only a partial check - for example, it does not check templates - so it is easily bypassed (nor should we trust it). This PR fixes the leak directly.

#### Which issue(s) this PR fixes or relates to

Contrib https://github.com/grafana/mimir/issues/73

#### Checklist

- [x] Tests updated.
- [x] Documentation added.
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [x] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: pkg/alertmanager/distributor_test.go [hunk 1]
```diff
@@ -330,6 +336,7 @@ func prepare(t *testing.T, numAM, numHappyAM, replicationFactor int, responseBod
 
 	return d, ams, func() {
 		require.NoError(t, services.StopAndAwaitTerminated(context.Background(), d))
+		require.NoError(t, services.StopAndAwaitTerminated(context.Background(), amRing))
```
Comment (alexweav) [inline pkg/alertmanager/distributor_test.go:339]: Drive-by fix: enabling the leak detector for `TestMultitenantAlertmanager_loadAndSyncConfigs` would fail due to unrelated goroutines leaked by this test. , This test only happens to fall earlier in the test ordering.

**Review by santihernandezc — APPROVED** [review]
LGTM 👍

**Review by 56quarters — APPROVED** [review]
LGTM, thanks!
